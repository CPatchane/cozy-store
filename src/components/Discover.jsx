import React, { Component } from 'react'

import { translate } from 'cozy-ui/react/I18n'

import SmallAppItem from './SmallAppItem'

const DEMO_APPS = [
  {
    slug: 'bank',
    name: 'Bank',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-bank.svg')
  },
  {
    slug: 'sante',
    name: 'Sant√©',
    developer: {name: 'MAIF Ltd'},
    editor: '',
    icon: require('../assets/temp_apps/icon-sante.svg')
  },
  {
    slug: 'mon_logis',
    name: 'Mon Logis',
    developer: {name: 'HoodBrains'},
    editor: '',
    icon: require('../assets/temp_apps/icon-mon_logis.svg')
  },
  {
    slug: 'calendar',
    name: 'Calendar',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-calendar.svg')
  }
]

class Discover extends Component {
  render () {
    const { t } = this.props
    return (
      <div className='sto-discover sto-soon'>
        <h2 className='sto-discover-title'>{t('discover.title')}</h2>
        <div className='sto-discover-get-started'>
          <h3 className='sto-discover-get-started-title'>
            {t('discover.get_started')}
          </h3>
          <div className='sto-discover-get-started-list'>
            {DEMO_APPS.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
              />
            })}
          </div>
        </div>
      </div>
    )
  }
}

export default translate()(Discover)
