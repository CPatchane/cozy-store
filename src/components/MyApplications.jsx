import React, { Component } from 'react'

import { Route } from 'react-router-dom'
import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import SmallAppItem from './SmallAppItem'
import ApplicationModal from './ApplicationModal'

class MyApplications extends Component {
  constructor (props) {
    super(props)
    props.fetchApps()
  }

  onAppClick (appSlug) {
    this.props.history.push(`/myapps/${appSlug}/manage`)
  }

  render () {
    const { t, myApps, isFetching, error } = this.props
    return (
      <div className='sto-myapps'>
        <h2 className='sto-myapps-title'>{t('myapps.title')}</h2>
        <div className='sto-myapps-list'>
          {!isFetching && myApps && !!myApps.length &&
            myApps.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
                onClick={() => this.onAppClick(a.slug)}
              />
            })
          }
          {error &&
            <p>{error}</p>
          }
        </div>
        {isFetching &&
          <Spinner
            size='xxlarge'
            loadingType='appsFetching'
            middle='true'
          />
        }

        <Route path='/myapps/:appSlug/manage' render={({ match }) =>
          (<ApplicationModal {...this.props} match={match} />)
        } />
      </div>
    )
  }
}

export default translate()(MyApplications)
