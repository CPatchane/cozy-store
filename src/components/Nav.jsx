import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import { NavLink } from 'react-router-dom'

export const Nav = ({ t }) => (
  <nav>
    <ul className='coz-nav'>
      <li className='coz-nav-item'>
        <NavLink to='/discover' className='sto-cat-discover coz-nav-link' activeClassName='active'>
          { t('nav.discover') }
        </NavLink>
      </li>
      <li className='coz-nav-item'>
        <NavLink to='/myapps' className='sto-cat-myapps coz-nav-link' activeClassName='active'>
          { t('nav.myapps') }
        </NavLink>
      </li>
    </ul>
  </nav>
)

export default translate()(Nav)
