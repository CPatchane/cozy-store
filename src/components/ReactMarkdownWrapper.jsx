import React from 'react'
import ReactMarkdown from 'react-markdown'

export const reactMarkdownRendererOptions = {
  Link: props => <a href={props.href} target='_blank'>{props.children}</a>
}

export const ReactMarkdownWrapper = ({ source }) =>
  <ReactMarkdown
    source={source}
    renderers={reactMarkdownRendererOptions}
  />

export default ReactMarkdownWrapper
