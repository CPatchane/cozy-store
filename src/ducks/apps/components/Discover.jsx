import React, { Component } from 'react'

import { Route } from 'react-router-dom'

import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import SmallAppItem from '../../components/SmallAppItem'
import UninstallModal from './UninstallModal'
import { InstallModal } from '../currentAppVersion/Containers'

export class Discover extends Component {
  constructor (props, context) {
    super(props)
    props.fetchApps(context.lang)

    this.onAppClick = this.onAppClick.bind(this)
  }

  onAppClick (appSlug) {
    this.props.history.push(`/discover/${appSlug}/manage`)
  }

  render () {
    const { t, apps, isFetching } = this.props
    return (
      <div className='sto-discover'>
        <h2 className='sto-discover-title'>{t('discover.title')}</h2>
        <div className='sto-discover-get-started'>
          <h3 className='sto-discover-get-started-title'>
            {t('discover.get_started')}
          </h3>
          <div className='sto-discover-get-started-list'>
            {!isFetching && apps.map(a => {
              // don't display app installed but not in registry
              if (!a.isInRegistry) return null
              let version = a.version
              if (a.versions && Array.isArray(a.versions.stable)) {
                version = (a.versions.stable && a.versions.stable[a.versions.stable.length - 1])
              }
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer || ''}
                editor={a.editor || ''}
                icon={a.icon}
                name={a.name}
                version={version}
                installed={a.installed}
                onClick={() => this.onAppClick(a.slug)}
              />
            })}
            {isFetching &&
              <Spinner
                size='xxlarge'
                loadingType='appsFetching'
                middle='true'
              />
            }
          </div>
        </div>
        <Route path='/discover/:appSlug/manage' render={({ match }) => {
          if (isFetching) return
          if (apps.length && match.params) {
            const app = apps.find(app => app.slug === match.params.appSlug)
            if (app.installed) {
              return <UninstallModal uninstallApp={this.props.uninstallApp} parent='/discover' app={app} match={match} />
            } else if (!app.installed) {
              return <InstallModal installApp={this.props.installApp} currentAppVersion={this.props.currentAppVersion} parent='/discover' app={app} match={match} />
            }
          }
        }} />
      </div>
    )
  }
}

export default translate()(Discover)
