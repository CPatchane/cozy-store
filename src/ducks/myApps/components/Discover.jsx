import React, { Component } from 'react'

import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import SmallAppItem from '../../components/SmallAppItem'

export class Discover extends Component {
  constructor (props) {
    super(props)
    props.fetchApps()
  }

  render () {
    const { t, apps, isFetching } = this.props
    return (
      <div className='sto-discover'>
        <h2 className='sto-discover-title'>{t('discover.title')}</h2>
        <div className='sto-discover-get-started'>
          <h3 className='sto-discover-get-started-title'>
            {t('discover.get_started')}
          </h3>
          <div className='sto-discover-get-started-list'>
            {!isFetching && apps.map(a => {
              const version = a.version || (a.versions && a.versions.stable && a.versions.stable[a.versions.stable.length - 1]) || ''
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer || ''}
                editor={a.editor || ''}
                icon={a.icon}
                name={a.name}
                version={version}
                installed={a.installed}
              />
            })}
            {isFetching &&
              <Spinner
                size='xxlarge'
                loadingType='appsFetching'
                middle='true'
              />
            }
          </div>
        </div>
      </div>
    )
  }
}

export default translate()(Discover)
