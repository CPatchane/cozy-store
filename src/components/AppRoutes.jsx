import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'

import MyApplications from '../containers/MyApplications'

const AppRoutes = () => {
  return <Switch>
    <Route path='/myapps' component={MyApplications} />
    <Redirect from='/' to='/myapps' />
  </Switch>
}

export default AppRoutes
