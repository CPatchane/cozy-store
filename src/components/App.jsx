
import React from 'react'

import Sidebar from './Sidebar'
import AppRoutes from './AppRoutes'

const App = () => (
  <div className='sto-wrapper coz-sticky'>
    <Sidebar />
    <main className='sto-content'>
      <AppRoutes />
    </main>
  </div>
)
export default App
