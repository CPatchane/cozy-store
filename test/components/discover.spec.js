'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { tMock } from '../jestLib/I18n'
import { Discover } from '../../src/ducks/components/Discover'

describe('Discover component', () => {
  it('should be rendered correctly', () => {
    const component = shallow(
      <Discover t={tMock} />
    ).node
    expect(component).toMatchSnapshot()
  })
})
