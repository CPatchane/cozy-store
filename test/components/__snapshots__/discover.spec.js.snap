// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discover component should be rendered correctly 1`] = `
<div
  className="sto-discover sto-soon"
  onClick={undefined}
>
  <h2
    className="sto-discover-title"
  >
    Discover
  </h2>
  <div
    className="sto-discover-get-started"
  >
    <h3
      className="sto-discover-get-started-title"
    >
      Get started with Cozy
    </h3>
    <div
      className="sto-discover-get-started-list"
    >
      <SmallAppItem
        developer={
          Object {
            "name": "Cozy",
          }
        }
        editor="Cozy"
        icon="test-file-stub"
        name="Bank"
        slug="bank"
        version={undefined}
      />
      <SmallAppItem
        developer={
          Object {
            "name": "MAIF Ltd",
          }
        }
        editor=""
        icon="test-file-stub"
        name="SantÃ©"
        slug="sante"
        version={undefined}
      />
      <SmallAppItem
        developer={
          Object {
            "name": "HoodBrains",
          }
        }
        editor=""
        icon="test-file-stub"
        name="Mon Logis"
        slug="mon_logis"
        version={undefined}
      />
      <SmallAppItem
        developer={
          Object {
            "name": "Cozy",
          }
        }
        editor="Cozy"
        icon="test-file-stub"
        name="Calendar"
        slug="calendar"
        version={undefined}
      />
    </div>
  </div>
</div>
`;
